src/components/Core.tsx

```diff
@@ -143,38 +142,43 @@ export class Core extends React.Component<IModalProps, IModalState> {
     }
   }
 
+  private validateCurrentChain () {
+    const { supportedChains } = this.props
+    const { chainId, provider } = this.state
+
+    const isCurrentChainSupported = supportedChains && supportedChains.includes(chainId!)
+
+    if (!isCurrentChainSupported) {
+      provider.on(CHAIN_CHANGED, () => this.setState({ currentStep: 'Step1' }))
+
+      // this message can be improved
```

Comment:
Soon, we should move all of the strings to a JSON file and then allow the developer to upload or include their own replacement strings.